jujugui/static/gui/src/app/store/env/bakery.js

```diff
@@ -0,0 +1,339 @@
+/*
+This file is part of the Juju GUI, which lets users view and manage Juju
+environments within a graphical interface (https://launchpad.net/juju-gui).
+Copyright (C) 2012-2015 Canonical Ltd.
+
+This program is free software: you can redistribute it and/or modify it under
+the terms of the GNU Affero General Public License version 3, as published by
+the Free Software Foundation.
+
+This program is distributed in the hope that it will be useful, but WITHOUT
+ANY WARRANTY; without even the implied warranties of MERCHANTABILITY,
+SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Affero
+General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License along
+with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+'use strict';
+
+/**
+ * Bakery holds the context for making HTTP requests
+ * that automatically acquire and discharge macaroons.
+ *
+ * @module env
+ * @submodule env.bakery
+ */
+
+YUI.add('juju-env-bakery', function(Y) {
+
+  var module = Y.namespace('juju.environments.web');
+  var macaroon = Y.namespace('macaroon');
+
+  /**
+   * Bakery client inspired by the equivalent GO code.
+   *
+   * This object exposes the ability to perform requests
+   * that automatically acquire and discharge macaroons
+   *
+   * @class Bakery
+   */
+
+  var Bakery = Y.Base.create('Bakery',
+      Y.Base, [], {
+
+      /**
+       Initialize.
+
+       @method initializer
+       @return {undefined} Nothing.
+       */
+      initializer: function () {
+        this.webhandler = new Y.juju.environments.web.WebHandler();
+      },
+
+      /**
+       Send get request.
+
+       @param {String} The path to make the api request to.
+       @param {String} The path to make a request to set the cookie.
+       @param {Function} successCallback Called when the api request completes
+       successfully.
+       @param {Function} failureCallback Called when the api request fails
+       with a response of >= 400 except 401/407 where it does authentication.
+       */
+      sendGetRequest: function (path, setCookiePath,
+                                successCallback, failureCallback) {
+        var macaroons = Y.Cookie.get('Macaroons');
```

Comment:
you mentioned that you were going to convert this away from YUI, did that turn out to be too much work for now?